core/src/main/java/org/apache/struts2/interceptor/parameter/ParametersInterceptor.java

```diff
@@ -400,8 +411,10 @@ protected boolean hasValidAnnotatedPropertyDescriptor(Object action, PropertyDes
             }
             return false;
         }
+        LOG.debug("Success: Matching annotated method [{}] found for property [{}] of depth [{}] on Action [{}]",
+                relevantMethod.getName(), propDesc.getName(), paramDepth, action.getClass().getSimpleName());
         if (paramDepth >= 1) {
-            allowlistClass(relevantMethod.getReturnType());
+            allowlistClass(propDesc.getPropertyType());
```

Comment:
Minor fix here to ensure the specific model type is allowlisted rather than the type declared in the interface (`Object.class`)